backend/services/implementations/userService.ts

```diff
@@ -333,6 +333,54 @@ class UserService implements IUserService {
       throw error;
     }
   }
+
+  /**
+   * This method gets all the users corresponding to the ids parameter
+   *
+   * @param ids the unique identifiers of the users to fetch
+   */
+  async findAllUsersByIds(ids: string[]): Promise<Array<UserDTO>> {
+    let userDtos: Array<UserDTO> = [];
+    try {
+      const users: Array<User> = await MgUser.find({ _id: { $in: ids } });
+
+      userDtos = await this.mapUsersToUserDTOs(users);
+    } catch (error: unknown) {
+      Logger.error(`Failed to get users. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return userDtos;
+  }
+
+  private async mapUsersToUserDTOs(
+    users: Array<User>,
+  ): Promise<Array<UserDTO>> {
+    const userDtos: Array<UserDTO> = await Promise.all(
+      users.map(async (user) => {
+        let firebaseUser: firebaseAdmin.auth.UserRecord;
+
+        try {
+          firebaseUser = await firebaseAdmin.auth().getUser(user.authId);
```

Comment:
Why do we error check the `user.authId` twice? Shouldn't any `authId` returned by the DB in line 345 be valid? Or is the second check just to reconcile differences between Firebase and DB? Not requesting a change, just curious!

Reply:
I believe it's the former. This logic already existed in starter-code in `getUsers`. I didn't update the existing code to use this logic because it doesn't have any unit test coverage.

Reply:
I believe it's the former. This logic already existed in starter-code in `getUsers`. I didn't update the existing code to use this logic because it doesn't have any unit test coverage.